### PR TITLE
[Feature] Upgrade SMCReader to support rigid transformation of SMPL

### PR DESCRIPTION
**Pull request by caizhongang**
- [x] Add rigid transformation for SMPL parameters to device frame
- [x] Support getting SMPL by device in SMCReader
- [x] Update HuMMan data convertion accordingly
- [x] Fix circular import
- [x] Add unit tests

### CODE REVIEW

File: mmhuman3d/data/data_structures/smc_reader.py [hunk 1]
```diff
@@ -942,36 +973,44 @@ def get_smpl(self,
...
+            torch_device = self.body_model.global_orient.device
+
+            assert device in {
+                'Kinect', 'iPhone'
+            }, f'Undefined device: {device}, should be "Kinect" or "iPhone"'
+            assert device_id >= 0
+
             if device == 'Kinect':
-                cam2world = self.get_kinect_color_extrinsics(
+                T_cam2world = self.get_kinect_color_extrinsics(
```
Comment (kimren227) [inline mmhuman3d/data/data_structures/smc_reader.py:988]: can we rename it to trans_cam2world?

File: mmhuman3d/models/body_models/utils.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    T_v2w = T_p2w @ T_v2p
+
+    # compute transformation from vertex to camera
+    T_v2c = T_w2c @ T_v2w
+
+    # decompose vertex to camera transformation
+    # p_: new pelvis frame
+    # T_v2c = T_p_2c x T_v2p_
+    T_p_2c = T_p2w
+    T_v2p_ = np.linalg.inv(T_p_2c) @ T_v2c
```
Comment (kimren227) [inline mmhuman3d/models/body_models/utils.py:51]: can we rename this to something like T_v2np? p_ is a bit ambiguous.

File: mmhuman3d/data/data_structures/smc_reader.py (hunk 1 above)
Comment (caizhongang) [reply mmhuman3d/data/data_structures/smc_reader.py:991]: `T_cam2world` is a (4, 4) transformation matrix, I think keeping the current name may be clearer as `trans_cam2world` may be understood as a (3, ) vector.

File: mmhuman3d/models/body_models/utils.py (hunk 2 above)
Comment (caizhongang) [reply mmhuman3d/models/body_models/utils.py]: Fixed in https://github.com/open-mmlab/mmhuman3d/pull/100/commits/13a42514f9eec927fce7412cc80d29dc4dbd88bd.

**Review by kimren227 — APPROVED** [review]

**Pull request merged**